src/server/project.ts

```diff
@@ -966,6 +970,10 @@ namespace ts.server {
                 info.detachFromProject(this);
             }
 
+            if (info.getRealpathIfDifferent()) {

```

Comment:
I think it would be better to take info as parameter in `onFileAddedOrRemoved` and do it there so that its always done consistently?

Reply:
That’s what exactly what I tried first, but that function isn’t called in a lot of cases! I found `this.hasAddedOrRemovedFiles` to be pretty unreliable in general. The problem is that it never gets called only when `info.detachFromProject()` gets called, so when the `detachFromProject` parameter here is false, this was the closest place I could get.

Reply:
Thats because you need to pick  dc6f9b428b360c49f443f643f90faab0cca4859d which was missed before.

Reply:
Ah perfect, thank you! Let me take a look.

Reply:
Fixed 👍